source/ocsp-support/ocsp-support.rst

```diff
@@ -238,11 +253,25 @@ per MongoClient basis.
 1. Driver MUST enable OCSP support (with stapling if possible) when
    certificate revocation checking is enabled.
 
-2. Drivers SHOULD throw an error if any of ``tlsInsecure=true`` or
+2. Drivers SHOULD throw an error if ``tlsInsecure=true`` or
    ``tlsAllowInvalidCertificates=true`` or
    ``tlsDisableOCSPEndpointCheck=true`` is specified alongside the
    option to enable certificate revocation checking.
 
+3. If a connection string contains both ``tlsInsecure`` and
+   ``tlsDisableCertificateRevocationCheck`` then the driver MUST throw
+   an error.
+
+4. If a driver supports ``tlsAllowInvalidCertificates`` and a
```

Comment:
I think we should keep that clause. If a driver doesn't support a URI option, I don't think it should have to record the value the user passed in just for a validation check that isn't really meaningful.

Reply:
I agree with Divjot's line of reasoning for keeping the clause. I can also see Sam's reasoning behind wanting to prevent users from accidentally configuring a client to be in a state where TLS contraints are different than they'd expect. On the flip side, if a driver did not support a given TLS URI option and the user passed it in, as of now, the result would appear to be more security than expected, since the optional TLS options currently only weaken security. (Additionally, I am uncertain how many drivers parse options and raise errors/warnings on options that they do not implement: if we do decide to force this behavior, we may need a drivers ticket).

For now, since I also agree that the wording should be consistent across the two specs, I have updated the URI options spec accordingly, noting that an error should be thrown if the driver supports said option.

Reply:
I pretty strongly disagree that the validation check isn't meaningful; if a user specifies `tlsAllowInvalidCertificates`, it seems fairly obvious they expect it to work, so proceeding rather than returning an error will at best cause them to be surprised by the behavior and at worst cause them not to understand the actual options that their application is using.

Reply:
Additionally, changing all of the existing rules in the URI options spec that aren't related to this PR at all to fit the phrasing here seems pretty dubious to me; if we want to make substantive non-OCSP related changes to the URI options spec (which the changes below are doing), we should file a spec ticket for it and deal with it separately.

Reply:
Hm, I believe I only changed the wording for the options relevant to the OCSP spec (tlsDisableOCSPEndpointCheck and tlsDisableCertificateRevocationCheck) in the URI options spec, so I don't think I've made non-ocsp related changes to the URI options spec? Happy to revert them if I have :)

Reply:
Hmm, looks like I misread; apologies for that. That said, the precedent in the URI options spec is still not to have this clause, so I think we should file a SPEC ticket to determine it one way or another for all of the rules in the URI options spec rather than having the OCSP rules be different.

Reply:
The URI options test plan also notes: "Note that there are tests for each of the options marked as optional; drivers will need to implement logic to skip over the optional tests that they don’t implement." 
Given that, I feel comfortable reverting the changes I made to the URI options test-plan. 

I'd also still advocate for keeping the clause in the OCSP support spec as it would be in line with what's stated in the URI-options test plan, and I'd be happy to strike the clause once drivers has decided how we want to handle the case of parsing non-implemented TLS options

Reply:
Fair enough. I'm honestly not sure which way it would go if up for discussion with the whole team, which is why I think a spec ticket to decide whether or not to add the clause to all those rules in the URI options spec would make sense.

Reply:
Okay, I've created https://jira.mongodb.org/browse/SPEC-1601 to clarify the behavior of conflicting TLS options and cc'ed you both.  As a reminder, I've also kept the clarifying clause in the OCSP spec for now and reverted the changes I made in the URI options test-plan stating that the errors only need to be thrown iff the driver supports all the options.  Sam, would you be comfortable resolving this thread given the above?

Reply:
Yes, this is fine